fastlane/Fastfile

```diff
@@ -125,7 +126,14 @@ private_lane :appstore_api_key do
 end
 
 lane :pod_lint do
-  pod_lib_lint(podspec: 'StreamChatSwiftUI.podspec', allow_warnings: true)
+  Dir.chdir('..') { sh("xcodebuild -resolvePackageDependencies -clonedSourcePackagesDirPath #{source_packages_path}") }
+  dependencies = Dir.chdir('..') { JSON.parse(File.read(project_package_resolved)) }
+  is_branch = dependencies['pins'].find { |dependency| dependency['identity'] == 'stream-chat-swift' && dependency['state']['branch'] }
+  if is_branch.nil?
+    pod_lib_lint(podspec: 'StreamChatSwiftUI.podspec', allow_warnings: true)
+  else
+    UI.important('Pod linting skipped: stream-chat-swift targets a development branch')
+  end
```

Comment:
Looks good, but would be great to avoid double `Dir.chdir`ing. What do you thing about something like this?
```suggestion
  lint_required = true
  Dir.chdir('..') do
    sh("xcodebuild -resolvePackageDependencies -clonedSourcePackagesDirPath #{source_packages_path}")
    dependencies = JSON.parse(File.read(project_package_resolved))
    lint_required = dependencies['pins'].none? { |dependency| dependency['identity'] == 'stream-chat-swift' && dependency['state']['branch'] }
  end

  if lint_required
    pod_lib_lint(podspec: 'StreamChatSwiftUI.podspec', allow_warnings: true)
  else
    UI.important('Pod linting skipped: stream-chat-swift targets a development branch')
  end
```